### PR TITLE
User guide tests for DNS certificate management

### PR DESCRIPTION
**Pull request by lei-tang**
Please provide a description for what this PR is for:
- Add user guide tests for DNS certificate management.
- Remove the user guide's dependency on jq.

### CODE REVIEW

File: tests/security/dns_cert/scripts/dns_cert.txt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+set -o pipefail
+
+source ${REPO_ROOT}/content/en/docs/tasks/security/dns-cert/snips.sh
+
+out=$(snip_check_the_provisioning_of_dns_certificates_1 2>&1)
+# Remove trailing spaces
+out=$(echo "$out" | sed 's/[ ]*$//')
+if [[ "$out" != *"$snip_check_the_provisioning_of_dns_certificates_2"* ]]; then
+    echo "FAILED snip_check_the_provisioning_of_dns_certificates_1:$out"; exit 1
+fi
```
Comment (frankbu) [inline tests/security/dns_cert/scripts/dns_cert.txt:28]: We've added some built-in [verification functions](https://github.com/istio/istio.io/blob/master/tests/util/verify.sh) to the framework. I think `_verify_contains` should work here.
```suggestion
_verify_contains "$out" "$snip_check_the_provisioning_of_dns_certificates_2" "snip_check_the_provisioning_of_dns_certificates_1"
```

File: tests/security/dns_cert/scripts/dns_cert.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+snip_regenerating_a_dns_certificate_1
+
+out=$(snip_regenerating_a_dns_certificate_2 2>&1)
+# Remove trailing spaces
+out=$(echo "$out" | sed 's/[ ]*$//')
+if [[ "$out" != *"$snip_regenerating_a_dns_certificate_3"* ]]; then
+    echo "FAILED snip_regenerating_a_dns_certificate_2:$out"; exit 1
+fi
```
Comment (frankbu) [inline tests/security/dns_cert/scripts/dns_cert.txt:37]:
```suggestion
_verify_contains "$out" "$snip_regenerating_a_dns_certificate_3" "snip_regenerating_a_dns_certificate_2"
```

File: tests/security/dns_cert/scripts/dns_cert.txt (hunk 1 above)
Comment (lei-tang) [reply tests/security/dns_cert/scripts/dns_cert.txt]: Done: the PR has been revised to use _verify_contains.

File: tests/security/dns_cert/scripts/dns_cert.txt (hunk 2 above)
Comment (lei-tang) [reply tests/security/dns_cert/scripts/dns_cert.txt]: Done: the PR has been revised to use _verify_contains.

File: tests/security/dns_cert/scripts/dns_cert.txt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+set -e
+set -u
+set -o pipefail
+
+source ${REPO_ROOT}/content/en/docs/tasks/security/dns-cert/snips.sh
+
+out=$(snip_check_the_provisioning_of_dns_certificates_1 2>&1)
+# Remove trailing spaces
+out=$(echo "$out" | sed 's/[ ]*$//')
```
Comment (frankbu) [inline tests/security/dns_cert/scripts/dns_cert.txt:25]: This also seems unnecessary. Does it fail if you remove?
```suggestion
```

File: tests/security/dns_cert/scripts/dns_cert.txt [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+_verify_contains "$out" "$snip_check_the_provisioning_of_dns_certificates_2" "snip_check_the_provisioning_of_dns_certificates_1"
+
+snip_regenerating_a_dns_certificate_1
+
+out=$(snip_regenerating_a_dns_certificate_2 2>&1)
+# Remove trailing spaces
+out=$(echo "$out" | sed 's/[ ]*$//')
```
Comment (frankbu) [inline tests/security/dns_cert/scripts/dns_cert.txt:32]:
```suggestion
```

File: tests/security/dns_cert/scripts/dns_cert.txt (hunk 3 above)
Comment (lei-tang) [reply tests/security/dns_cert/scripts/dns_cert.txt:25]: This is necessary and the test fails if this line is removed.

File: tests/security/dns_cert/scripts/dns_cert.txt (hunk 4 above)
Comment (lei-tang) [reply tests/security/dns_cert/scripts/dns_cert.txt:32]: This line (i.e., removing the trailing spaces in $out) is necessary and the test fails if this line is removed.

File: tests/security/dns_cert/scripts/dns_cert.txt (hunk 3 above)
Comment (frankbu) [reply tests/security/dns_cert/scripts/dns_cert.txt:25]: Hmm, it is hard to understand why. Maybe there's a better way to handle this case, but let's merge it as is for now and maybe revisit later.

**Pull request merged**